docs/source/notes/amp_examples.rst

```diff
@@ -0,0 +1,158 @@
+.. _amp-examples:
+
+Automatic Mixed Precision Examples
+==================================
+
+.. currentmodule:: torch.cuda.amp
+
+.. contents:: :local:
+
+.. _gradient-scaling-examples:
+
+Gradient Scaling
+^^^^^^^^^^^^^^^^
+
+Gradient scaling helps ensure convergence when training with mixed precision,
+as explained :ref:`here<gradient-scaling>`.
+
+The following code snippets show how :class:`torch.cuda.amp.AmpScaler` can be used to
+perform dynamic gradient scaling automatically.
+
+Working with a Single Optimizer
+-------------------------------
+
+::
+
+    # Create an AmpScaler instance.
+    scaler = AmpScaler()
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+
+        # Scale the loss, and call backward() on the scaled loss to create scaled gradients.
+        scaler.scale(loss).backward()
+
+        # Carry out a scaling-safe step.  scaler.step() unscales the optimizer's gradients
+        # and skips optimizer.step() if the gradients contain infs or NaNs.
+        scaler.step(optimizer)
+
+        # Update the scale for next iteration.
+        scaler.update()
+
+
+Working with Unscaled Gradients
+-------------------------------
+
+The gradients resulting from ``scaler.scale(loss).backward()`` are scaled.  ``scaler.unscale`` can be
+used to unscale an optimizer's owned gradients in-place prior to ``scaler.step``.
+
+Gradient clipping
+"""""""""""""""""
+
+Gradient clipping involves manipulating a set of gradients such that their global norm
+(see :func:`torch.nn.utils.clip_grad_norm_`) or maximum magnitude (see :func:`torch.nn.utils.clip_grad_value_`)
+is :math:`<=` some user-imposed threshold.  Calling ``scaler.unscale(optimizer)`` before clipping enables you
+to clip unscaled gradients as usual::
+
+    scaler = AmpScaler()
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+        scaler.scale(loss).backward()
+
+        # Unscale the gradients owned by optimizer in-place
```

Comment:
Add comment here that unscaling is now done and the step that follows no longer unscales (again). 

Natalia: this comment should go before the step line, not here.
Greg: I agree with Natalia.